app.py

```diff
@@ -129,3 +145,19 @@ def display_template(template_slug):
                 matching_templates, name, postcode, address
             )
             return render_template("single_email.html", email=email_template[0])
+
+
+@app.route("/newsletter", methods=["POST"])
+def subscribe_to_newsletter():
+
+    post_params = {"email_address": request.form["email"], "status": "subscribed"}
+    r = requests.post(
+        url, auth=("foo", os.environ["MAILCHIMP_SECRET_KEY"]), json=post_params
```

Comment:
Do we maybe want to change the user to something other than "foo" (guessing that's what is done here)? If it doesn't matter then don't worry about it.

Reply:
the username here used for authentication with mailchimp api  is completely optional so I just added something random.